org.eclipse.scanning.server/src/org/eclipse/scanning/server/servlet/ScanProcess.java

```diff
@@ -226,31 +228,43 @@ private void validateRequest(ScanBean bean) throws ValidationException, Instanti
 	 * @throws Exception
 	 */
 	private void checkMonitors(IPointGenerator<?> gen) throws Exception {
-		Collection<String> monitorNames = bean.getScanRequest().getMonitorNames();
+		Collection<String> monitorNamesPerPoint = bean.getScanRequest().getMonitorNamesPerPoint();
+		Collection<String> monitorNamesPerScan = bean.getScanRequest().getMonitorNamesPerScan();
 		
 		// We set any activated monitors in the request if none have been specified.
 		if (Boolean.getBoolean("org.eclipse.scanning.server.useDefaultActivatedMonitors")) {
-			if (bean.getScanRequest().getMonitorNames()==null) {
-				monitorNames = getMonitors();
-				logger.debug("Using default monitors: {}", monitorNames);
+			if (monitorNamesPerPoint == null) {
+				monitorNamesPerPoint = getMonitors(MonitorRole.PER_POINT);
+				logger.debug("Using default per point monitors: {}", monitorNamesPerPoint);
+			}
+			if (monitorNamesPerScan == null) {
+				monitorNamesPerScan = getMonitors(MonitorRole.PER_SCAN);
+				logger.debug("Using default per scan monitors: {}", monitorNamesPerScan);
 			}
 		}
 		
-		if (monitorNames != null) {
+		if (monitorNamesPerPoint != null) {
+			// remove any monitors
+			Collection<String> scannableNames = getScannableNames(gen);
+			monitorNamesPerPoint = monitorNamesPerPoint.stream().filter(mon -> !scannableNames.contains(mon)).collect(Collectors.toList());
+					
+			bean.getScanRequest().setMonitorNamesPerPoint(monitorNamesPerPoint);
+		}
+		if (monitorNamesPerScan != null) {
 			// remove any monitors
 			Collection<String> scannableNames = getScannableNames(gen);
-			monitorNames = monitorNames.stream().filter(mon -> !scannableNames.contains(mon)).collect(Collectors.toList());
+			monitorNamesPerScan = monitorNamesPerScan.stream().filter(mon -> !scannableNames.contains(mon)).collect(Collectors.toList());
 					
-			bean.getScanRequest().setMonitorNames(monitorNames);
+			bean.getScanRequest().setMonitorNamesPerScan(monitorNamesPerScan);
 		}
 	}
 
-	private Collection<String> getMonitors() throws Exception {
+	private Collection<String> getMonitors(MonitorRole monitorRole) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)